cmd/automountServiceAccountToken_test.go

```diff
@@ -28,11 +28,11 @@ func TestDeploymentASAT(t *testing.T) {
 	}
 
 	for _, result := range results {
-		if result.name == "fakeDeploymentASAT1" && (result.err != 3 || result.dsa == "") {
+		if result.name == "fakeDeploymentASAT1" && (result.err != ErrorServiceAccountTokenDeprecated || result.dsa == "") {
```

Comment:
This result.dsa == "" could go away and instead you can add another test for result.sa == ""